contracts/colony/ColonyRoles.sol

```diff
@@ -103,7 +104,14 @@ contract ColonyRoles is ColonyStorage {
         setTo = uint256(roles) % 2 == 1;
 
         ColonyAuthority(address(authority)).setUserRole(_user, _domainId, roleId, setTo);
-
+        if (roleId == 0){
+          if (setTo){
```

Comment:
I don't think this quite gets us there, actually, since it doesn't check if the user already has the role or not. If I already have the recovery role, and then I use `setUserRoles` to give it to myself again, then this number will be bumped twice. We'd want something like

```
if (setTo && !hasUserRole(user, recovery))
```

Reply:
I disagree. This is inside the `if (changed)` statement, which stops that, no?

Reply:
Either way, testing for this wouldn't go amiss

Reply:
Ah right, that bit was clipped out of the Github viewer. I agree then. My other request is that we use `uint8(ColonyRoles.Recovery)` instead of `0` in the external conditional.